src/runtime.js

```diff
@@ -67,6 +68,11 @@ export function adopt(global) {
         name: name,
         implementationClass: implementationClass
       });
+      // Resolve this extension's Service Promise.
+      getService(global, name, () => {
```

Comment:
Could you please clarify what it means? Do we emulate an extension as a service here? Did you check that there are no name conflicts?